crates/wasi-common/src/ctx.rs

```diff
@@ -336,15 +339,16 @@ impl WasiCtxBuilder {
             log::debug!("WasiCtx inserting entry {:?}", pending);
             let fd = match pending {
                 PendingEntry::Thunk(f) => {
-                    let handle = EntryHandle::new(f()?);
-                    let entry = Entry::from(handle)?;
+                    let handle = EntryHandle::from(f()?);
+                    let entry = Entry::new(handle);
                     entries
                         .insert(entry)
                         .ok_or(WasiCtxBuilderError::TooManyFilesOpen)?
                 }
                 PendingEntry::File(f) => {
-                    let handle = EntryHandle::new(OsHandle::from(f));
-                    let entry = Entry::from(handle)?;
+                    let handle = OsOther::try_from(f)?;
```

Comment:
Why does this use `OsOther` here? Looking at the code, that seems to be for things that aren't files.

Reply:
Oh, see my reply to your comment about `Stdio` :-)

Reply:
In order to clear it up a little bit, I've renamed `PendingEntry::File` to `PendingEntry::OsHandle`. This naming was somewhat confusing since the `File` in this case represented for instance a pipe which we then used in place of a stdio handle. Oh and this is precisely why we use `OsOther` here instead of an `OsFile`, since the former is meant to represent pipes as well, whereas the latter only regular files.